src/API.jl

```diff
@@ -907,16 +907,17 @@ function _is_stale(paths::Vector{String}, sourcepath::String)
     return true
 end
 
-function _auto_precompile(ctx::Context; kwargs...)
+function _auto_precompile(ctx::Context)
     if parse(Int, get(ENV, "JULIA_PKG_PRECOMPILE_AUTO", "1")) == 1
-        Pkg.precompile(ctx; internal_call=true, kwargs...)
+        Pkg.precompile(ctx; internal_call=true)
     end
 end
 
 precompile(; kwargs...) = precompile(Context(); kwargs...)
-function precompile(ctx::Context; internal_call::Bool=false, io::IO=stderr)
+function precompile(ctx::Context; internal_call::Bool=false)
```

Comment:
Still needs `io` though? But it shoule be consumed by the context

Reply:
Yeah, tweaked it slightly.